django/contrib/admin/utils.py

```diff
@@ -74,6 +74,16 @@ def prepare_lookup_value(key, value, separator=","):
     return value
 
 
+def build_q_object_from_lookup_parameters(parameters):
+    q_object = models.Q()
+    for param, param_item_list in parameters.items():
+        param_q_obj = models.Q(**{param: param_item_list[0]})
+        for item in param_item_list[1:]:
+            param_q_obj |= models.Q(**{param: item})
+        q_object &= param_q_obj
```

Comment:
Right I thought about it and I was going to have something like this:
```
def build_q_object_from_lookup_parameters(parameters):
    if not parameters:
        return models.Q()
    field_q_objects = {}
    for param, param_item_list in parameters.items():
        field = param.split("__")[0]
        q_object = reduce(or_, (models.Q((param, item)) for item in param_item_list))
        if field in field_q_objects:
            field_q_objects[field] = field_q_objects[field] | q_object
        else:
            field_q_objects[field] = q_object
    return reduce(and_, field_q_objects.values())
```
But it breaks a bunch of tests because we have things like date ranges with `{"date_registered__gte": self.today, "date_registered__lt": self.tomorrow}` which were previously an `&` and this makes them an `|`

So I think different lookups in the same filter should be not be OR-ed

Reply:
Agreed.